src/api/block_builder/input_selection/utxo_chains/mod.rs

```diff
@@ -47,13 +47,19 @@ pub(crate) fn select_utxo_chain_inputs(
     // We track here for which outputs we did that, to prevent doing it multiple times.
     let mut added_output_for_input_signing_data = HashSet::new();
 
+    let mut required_alias_nft_addresses = HashSet::new();
+
     // Add existing selected inputs we added for sender and issuer features before
     for input_signing_data in selected_inputs.iter() {
         // Add inputs to outputs if not already there, so they don't get burned
         if !allow_burning {
             add_output_for_input(input_signing_data, rent_structure, outputs, token_supply)?;
         }
         added_output_for_input_signing_data.insert(input_signing_data.output_id()?);
+        let address = Address::try_from_bech32(&input_signing_data.bech32_address)?.1;
+        if address.is_alias() || address.is_nft() {
+            required_alias_nft_addresses.insert(address);
+        }
```

Comment:
Just checking, not doing this for the inputs we add later to `selected_inputs` ?